### PR TITLE
[SPARK-41077][CONNECT][PYTHON][REFACTORING] Rename `ColumnRef` to `Column` in Python client implementation

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

Connect python client uses `ColumnRef` to represent columns in API (e.g. `df.name`). Current PySpark uses `Class Column` for the same thing. In this case, we can align Connect with PySpark, which can help existing PySpark users to reuse their code for Spark Connect python client as much as possible (minimize the code change).

### Why are the changes needed?

This is to help existing PySpark users to reuse their code for Spark Connect python client as much as possible (minimize the code change).

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

Existing UT

### CODE REVIEW

**Comment by amaliujia** [comment]
R: @HyukjinKwon

File: python/pyspark/sql/connect/column.py [hunk 1]
```diff
@@ -30,8 +30,8 @@
 
 def _bin_op(
     name: str, doc: str = "binary function", reverse: bool = False
-) -> Callable[["ColumnRef", Any], "Expression"]:
-    def _(self: "ColumnRef", other: Any) -> "Expression":
+) -> Callable[["Column", Any], "Expression"]:
+    def _(self: "Column", other: Any) -> "Expression":
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/column.py:34]: I think this is a bit unclear. For example,  Spark Connect's `Expression` has binary operators like `pyspark.sql.Column` does. If Spark Connect's `ColumnRef` indicates a column reference only, it should be something like `AttributeReference`. cc @cloud-fan

Comment (amaliujia) [reply python/pyspark/sql/connect/column.py:34]: yeah I was stopped several times when implementing API by current expressions in Python side. My current focus is to improve here.....

Comment (amaliujia) [reply python/pyspark/sql/connect/column.py:34]: This Python `column` is to match the Scala `Column` I believe: https://github.com/apache/spark/blob/865a3ded2ea1ca86be93df58205882bc509b98cd/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L151

Comment (HyukjinKwon) [reply python/pyspark/sql/connect/column.py:34]: `Expression` is basically a wrapper for `Column`

Comment (HyukjinKwon) [reply python/pyspark/sql/connect/column.py:34]: cc @grundprinzip too FYI

Comment (amaliujia) [reply python/pyspark/sql/connect/column.py:34]: I am sure the current `Column` (or `ColumnRef`) is a concept of  reference attribute based on its `to_plan` impl:

```
    def to_plan(self, session: Optional["RemoteSparkSession"]) -> proto.Expression:
        """Returns the Proto representation of the expression."""
        expr = proto.Expression()
        expr.unresolved_attribute.unparsed_identifier = self._unparsed_identifier
        return expr
```

Comment (amaliujia) [reply python/pyspark/sql/connect/column.py:34]: Class comment says the same:

```
class ColumnRef(Expression):
    """Represents a column reference. There is no guarantee that this column
    actually exists. In the context of this project, we refer by its name and
    treat it as an unresolved attribute. Attributes that have the same fully
    qualified name are identical"""
 ```

Comment (amaliujia) [reply python/pyspark/sql/connect/column.py:34]: Ok I guess my question is do we want to match what PySpark is doing? PySpark pretty much uses `Column` as `Expression`, which is exposed on API surface.

Connect is exposing this `ColumnRef` to the API surface but only meaning for unresolved attribute reference. 

My initial thought for this PR is to at least matching what PySpark is doing. Rename first, then expand this class to match PySpark.

**Comment by cloud-fan** [comment]
In the Scala DataFrame, `Column` is the entry API for users to build an expression tree. For example, `df.select($"a".cast("int"))`, `$"a"` is a syntax sugar to create a `Column` with underlying expression `UnresolvedAttribute(Seq("a"))`.

**Comment by amaliujia** [comment]
@HyukjinKwon yes the goal will be matching the API shape of the `Column` in Python/Scala (likely Python first if there is API difference between Python and Scala).

This PR is enforce DataFrame API interface uses `Column` which matches PySpark. For next a few PR I will refactor to promote this class as the top level `Expression` class (how many PRs probably depends on the scope and complexity).

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**